src/pytesmo/validation_framework/upscaling.py

```diff
@@ -0,0 +1,391 @@
+
+
+class MixinReadTs:
+    """Mixin class to provide the reading function in DataAverager and DataManager"""
+
+    def read_ds(self, name, *args):
+        """
+        Function to read and prepare a datasets.
+
+        Calls read_ts of the dataset.
+
+        Takes either 1 (gpi) or 2 (lon, lat) arguments.
+
+        Parameters
+        ----------
+        name : string
+            Name of the other dataset.
+        args: either gpi or (lon, lat)
+            * gpi (int): Grid point index
+            * lon (float): Longitude of point
+            * lat(float): Latitude of point
+
+        Returns
+        -------
+        data_df : pandas.DataFrame or None
+            Data DataFrame.
+
+        """
+        ds = self.datasets[name]
+        args = list(args)
+        args.extend(ds['args'])
+
+        try:
+            func = getattr(ds['class'], self.read_ts_names[name])
+            data_df = func(*args, **ds['kwargs'])
+            if type(data_df) is TS or issubclass(type(data_df), TS):
+                data_df = data_df.data
+        except IOError:
+            warnings.warn(
+                "IOError while reading dataset {} with args {:}".format(name,
+                                                                        args))
+            return None
+        except RuntimeError as e:
+            if e.args[0] == "No such file or directory":
+                warnings.warn(
+                    "IOError while reading dataset {} with args {:}".format(name,
+                                                                            args))
+                return None
+            else:
+                raise e
+
+        if len(data_df) == 0:
+            warnings.warn("No data for dataset {}".format(name))
+            return None
+
+        if not isinstance(data_df, pd.DataFrame):
+            warnings.warn("Data is not a DataFrame {:}".format(args))
+            return None
+
+        if self.period is not None:
+            # here we use the isoformat since pandas slice behavior is
+            # different when using datetime objects.
+            data_df = data_df[
+                      self.period[0].isoformat():self.period[1].isoformat()]
+
+        if len(data_df) == 0:
+            warnings.warn("No data for dataset {} with arguments {:}".format(name,
+                                                                             args))
+            return None
+
+        else:
+            return data_df
+
+
+class Upscaling(MixinReadTs):
+    """
+    This class provides methods to combine the measurements of validation datasets (others) that fall under the same
+    gridpoint of the dataset being validated (reference).
+
+    The goal is to include here all identified upscaling methods to provide an estimate at the reference footprint scale.
+
+    Implemented methods:
+
+        * time-stability filtering
+        * simple averaging
+
+    Parameters
+    ----------
+    ref_class: pygeogrids Grid obj.
```

Comment:
There should be a space also before the colon, otherwise the documentation won't render correctly. I think you need to change this in all docstrings.